c7n/filters/core.py

```diff
@@ -449,7 +449,7 @@ def validate(self):
             if not self.data['op'] in OPERATORS:
                 raise PolicyValidationError(
                     "Invalid operator in value filter %s" % self.data)
-            if self.data['op'] in {'regex', 'regex-case'}:
+            if self.data['op'] in {'regex', 'regex-case'} and self.data.get('value'):
```

Comment:
how does the additional conditional enable the behavior of the pr title? it looks more like an additional safety check?

Reply:
Without it you cannot do thinks like..
Cause it tries to get value and test if regexp complies..

```
          - type: value
            key: zone
            op: regex
            value_from:
              url: http://give_me_allowed_regions_for_project&project={account_id}
              format: json
              expr: allowed_zones_regexp
```